code/modules/paperwork/paper.dm

```diff
@@ -108,220 +147,87 @@
-
-
-/obj/item/paper/Topic(href, href_list)
-	..()
-	var/literate = usr.is_literate()
-	if(!usr.canUseTopic(src, BE_CLOSE, literate))
-		return
-
-	if(href_list["help"])
-		openhelp(usr)
-		return
-	if(href_list["write"])
-		var/id = href_list["write"]
-		var/t =  stripped_multiline_input("Enter what you want to write:", "Write", no_trim=TRUE)
-		if(!t || !usr.canUseTopic(src, BE_CLOSE, literate))
-			return
-		var/obj/item/i = usr.get_active_held_item()	//Check to see if he still got that darn pen, also check if he's using a crayon or pen.
-		var/iscrayon = 0
-		if(!istype(i, /obj/item/pen))
-			if(!istype(i, /obj/item/toy/crayon))
-				return
-			iscrayon = 1
-
-		if(!in_range(src, usr) && loc != usr && !istype(loc, /obj/item/clipboard) && loc.loc != usr && usr.get_active_held_item() != i)	//Some check to see if he's allowed to write
-			return
-
-		log_paper("[key_name(usr)] writing to paper [t]")
-		t = parsepencode(t, i, usr, iscrayon) // Encode everything from pencode to html
-
-		if(t != null)	//No input from the user means nothing needs to be added
-			if(id!="end")
-				addtofield(text2num(id), t) // He wants to edit a field, let him.
-			else
-				info += t // Oh, he wants to edit to the end of the file, let him.
-				updateinfolinks()
-			usr << browse("<HTML><HEAD><TITLE>[name]</TITLE></HEAD><BODY>[info_links]<HR>[stamps]</BODY><div align='right'style='position:fixed;bottom:0;font-style:bold;'><A href='?src=[REF(src)];help=1'>\[?\]</A></div></HTML>", "window=[name]") // Update the window
-			update_icon_state()
+/obj/item/paper/can_interact(mob/user)
+	if(!..())
+		return FALSE
+	if(resistance_flags & ON_FIRE)		/// Are we on fire?  Hard ot read if so
+		return FALSE
+	if(user.is_blind())					/// Even harder to read if your blind...braile? humm
+		return FALSE
+	return user.can_read(src)			// checks if the user can read.
 
 
 /obj/item/paper/attackby(obj/item/P, mob/living/carbon/human/user, params)
-	..()
-
-	if(resistance_flags & ON_FIRE)
-		return
-
-	if(user.is_blind())
-		return
-
 	if(istype(P, /obj/item/pen) || istype(P, /obj/item/toy/crayon))
-		if(user.is_literate())
-			user << browse("<HTML><HEAD><TITLE>[name]</TITLE></HEAD><BODY>[info_links]<HR>[stamps]</BODY><div align='right'style='position:fixed;bottom:0;font-style:bold;'><A href='?src=[REF(src)];help=1'>\[?\]</A></div></HTML>", "window=[name]")
+		if(!form_sheet && length(info) >= 1000) // Sheet must have less than 1000 charaters
+			to_chat(user, "<span class='warning'>This sheet of paper is full!</span>")
 			return
-		else
-			to_chat(user, "<span class='warning'>You don't know how to read or write!</span>")
+		if(edit_mode != MODE_READING)
+			to_chat(user, "<span class='warning'>[edit_usr.real_name] is already working on this sheet!</span>")
 			return
 
+		edit_mode = MODE_WRITING
+		edit_usr = user
+		/// should a crayon be in the same subtype as a pen?  How about a brush or charcoal?
+		/// TODO:  Convert all writing stuff to one type, /obj/item/art_tool maybe?
```

Comment:
Not the biggest of deals so you don't have to fix this, but keep inline comments to `//` in the future to avoid autodoc confusion.

Reply:
Ah ok,  Started to be a habbit